### PR TITLE
fix(core): slug overwrite

### LINKED ISSUE #1164: Prevent overwriting existing file
**- Do you want to request a *feature* or report a *bug*?**
BUG

**- What is the current behavior?**
When I add a file with the same title as an existing one and my slug equals title, the existing file is overwritten.

**- If the current behavior is a bug, please provide the steps to reproduce.**
make sure your slug equals title. add a file, choose a title that is already used by another file, save.
you have overwritten the existing file 

**- What is the expected behavior?**
- either: 
a new file is created (i.e. with counter to filename)
- or:
a warning that the file cannot be saved as it needs a unique title

Less urgent if #1069 were implemented.

### PR DESCRIPTION
**Pull request by barthc**
**Summary**
Closes #1164, fix  extracted  from #1931

### CODE REVIEW

File: packages/netlify-cms-core/src/actions/entries.js [hunk 1]
```diff
@@ -452,6 +452,7 @@ export function persistEntry(collection) {
     const state = getState();
     const entryDraft = state.entryDraft;
     const fieldsErrors = entryDraft.get('fieldsErrors');
+    const usedSlugs = selectAllSlugEntries(state, collection.get('name'));
```
Comment (erquhart) [inline packages/netlify-cms-core/src/actions/entries.js]: I could be wrong, but I don't believe `persistEntry` is used when the editorial workflow is active, so selecting just the slugs for published entries should suffice. Did you find that not to be the case?

**Comment by barthc** [comment]
@erquhart  another look pls.

**Comment by erquhart** [comment]
@barthc we'll get this into today's beta release 👍

**Comment by barthc** [comment]
@erquhart thanks for merging this, really appreciate it 👍

**Comment by erquhart** [comment]
No problem, thanks for getting it in! It's now in the latest beta release `npm install netlify-cms@beta`.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**